src/borg/archiver/debug_cmd.py

```diff
@@ -518,6 +574,72 @@ def build_parser_debug(self, subparsers, common_parser, mid_common_parser):
             "path", metavar="PATH", type=str, help="content for which the id-hash shall get computed"
         )
 
+        # parse_obj
+        debug_parse_obj_epilog = process_epilog(
+            """
+                This command parses the object file into metadata (as json) and uncompressed data.
+                """
+        )
+        subparser = debug_parsers.add_parser(
+            "parse-obj",
+            parents=[common_parser],
+            add_help=False,
+            description=self.do_debug_parse_obj.__doc__,
+            epilog=debug_parse_obj_epilog,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            help="parse objectfile into meta dict and data",
+        )
+        subparser.set_defaults(func=self.do_debug_parse_obj)
+        subparser.add_argument("id", metavar="ID", type=str, help="hex object ID to get from the repo")
+        subparser.add_argument(
+            "cdata", metavar="COMPRESSED_DATA", type=str, help="path of the objectfile to parse data from"
+        )
+        subparser.add_argument(
+            "data_path", metavar="DATA_PATH", type=str, help="path of the file to write uncompressed data into"
+        )
+        subparser.add_argument(
+            "json_path", metavar="JSON_PATH", type=str, help="path of the json file to write metadata into"
+        )
+
+        # format_obj
+        debug_format_obj_epilog = process_epilog(
+            """
+                This command formats the file and metadata into objectfile.
+                """
+        )
+        subparser = debug_parsers.add_parser(
+            "format-obj",
+            parents=[common_parser],
+            add_help=False,
+            description=self.do_debug_format_obj.__doc__,
+            epilog=debug_format_obj_epilog,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            help="format file and metadata into borg objectfile",
+        )
+        subparser.set_defaults(func=self.do_debug_format_obj)
+        subparser.add_argument("id", metavar="ID", type=str, help="hex object ID to get from the repo")
+        subparser.add_argument(
+            "data_path", metavar="DATA_PATH", type=str, help="path of the file to convert into objectfile"
+        )
+        subparser.add_argument(
+            "json_path", metavar="JSON_PATH", type=str, help="path of the json file to read metadata from"
+        )
```

Comment:
consistency:
- if the first is DATA_PATH, the second needs to be META_PATH
- otherwise you would refer to what's inside for the first and to the specific file format for the second.

(please globally check this)

Reply:
also, it would be better for intuition / remembering the order if argument order was 1. meta 2. data.

or, as a completely different approach:

just give one path as cli argument and append `.json` for the metadata and `.bin` for the data file.

Reply:
My logic was that we are providing two different file-type (so `some_kind_of_data` and `json`) instead of two different file-contents (`data` and `meta`) ... so I thought that `json` will make sense.

Also about the second approach, I thought that explicitly mentioning the filepaths will be much better if someone was trying to debug stuff, since they won't have to constantly rename things to be the same for the command to run. Though I guess if `json` is being generated by some piece of code that the user is using, it will be not be a problem.

Reply:
ah, so you thought as data as a generalisation of binary or text or whatever. i thought data as opposite of meta.

considering all formats can also be interpreted as binary, even if they are text or some other specific format, you could use **binary** for what you meant.